ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+    # Here we delegate the write method to save the files to different helper functions
+    # For different formats, we can define different helper functions, and based on the file type
+    # we can call them. Currently only FITS is supported.
+
+    # Here we create a constructor for a mixin class, which is quite rare for a mixin
+    # Since the data of the NDCube needs to be used in this mixin, the
+    # constructor seems to be useful
+
+    def __init__(self, data, wcs, uncertainty, mask, meta,
+                 unit, copy, extra_coords=None, missing_axis=None, **kwargs):
+
+        super().__init__(data, wcs=wcs, uncertainty=uncertainty, mask=mask,
+                         meta=meta, unit=unit, copy=copy, **kwargs)
+        self.__data = data
+        self.__wcs = wcs
+        self.__meta = meta
+        self.__uncertainty = super().uncertainty
+        self.__uncertainty_type = None
+        self.__mask = mask
+        self.__unit = unit
+        self.__missing_axis = missing_axis
+        self.__extra_coords = extra_coords
+
+
+
+    def to_hdu(self, hdu_mask='MASK', hdu_uncertainty='UNCERT'):
+        """Create a HDUList from a ImageHDU object
+
+        Parameters
+        ----------
+        hdu_mask, hdu_uncertainty : str, optional
+            If it is a string append this attribute to the HDUList as
+            `astropy.io.fits.ImageHDU` with the string as the extension name.
+            Default is `MASK` for hdu_mask, `UNCERT` for uncertainty and `None`
+            for flags.
+
+
+        Raises
+        ------
+        ValueError
+            - If `self.__mask` is set but not a `numpy.ndarray`.
+            - If `self.__uncertainty` is set but not a astropy uncertainty type
+            - If `self.__uncertainty` is set but has another unit than `self.__data`.
+
+        Returns
+        -------
+        hdulist:
+            `astropy.io.fits.HDUList`
+        """
+
+
+        # ----------------------------------HDU0------------------------------
+        # Create a copy of the meta data to avoid changing of the header of data
+        if self.__meta is not None:
+            header = fits.Header(self.__meta.copy())
+
+        else:
+            header = fits.Header()
+        # Create a FITS Safe header from the meta data
+
+        # Cross-check if the metadata is fits safe, if not then it is corrected
+        for k, v in header.items():
+            _insert_in_metadata_fits_safe(header, k, v)
+
+        if self.__wcs:
+            # Create a header for a given wcs object
+            # Hard-Coded relax parameter to write all
+            # recognized informal extensions of the WCS standard.
+            wcs_header = self.__wcs.to_header(relax=True)
+            header.extend(wcs_header, useblanks=False, update=True)
+
+        # Create a FITS header for storing missing_axis
+        # MISSING0 : 1 if axis 1 is missing
+        # other keywords are skipped, if the axis is present
+        if self.__missing_axis:
```

Comment:
I don't think this line is needed as an ```NDCube``` will always have a ```missing_axis``` attribute that is a list of ```bool```, even if all ```bool```s are ```False```.

Reply:
Yeah, this seems redundant.